esp/src/eclwatch/DFUQueryWidget.js

```diff
@@ -377,9 +379,31 @@ define([
                     EndDate: registry.byId(this.id + "ToDate").attr("value").toISOString().replace(/T.*Z/, '') + "T23:59:59Z"
                 });
             }
+           
+            this.updatedFilter = retVal;
+            this.checkIfWarning();
+           
             return retVal;
         },
 
+        checkIfWarning: function () {
+            var context = this;
+
+            WsDfu.DFUQuery({
+                request: context.updatedFilter
+            }).then(function (response){
+                if (lang.exists("DFUQueryResponse", response)) {
+                    if (response.DFUQueryResponse.Warning) {
+                        context.filter.open();
+                        context.filter.setFilterMessage(context.i18n.FilesWarning);
+                    } else {
+                        context.filter.close();
```

Comment:
Should we be closing the filter dialog at this point?

Reply:
Yes, we have to close the filter because if the user navigates to another widget at this point the filter persists throughout different widgets.